feat: OPTIC-2161: Give Select a way to override the display value completely

yyassi-heartex opened the pull request:
This pull request introduces a new `displayValueOverride` prop to the `Select` component, allowing developers to customize how selected values are displayed. It also includes updates to the `Select` component and its associated stories to demonstrate this new functionality.

<img width="566" alt="Screenshot 2025-05-06 at 1 18 42 PM" src="https://github.com/user-attachments/assets/88d233aa-1e64-436a-9ac8-1051365f3b1f" />

### Enhancements to `Select` Component:

* **Added `displayValueOverride` Prop**: A new optional prop, `displayValueOverride`, was introduced to allow custom rendering of selected values in the `Select` component. This prop takes a function that receives `selectedOptions` and an optional `placeholder`, and returns a custom display value. (`web/libs/ui/src/lib/select/types.ts` - [[1]](diffhunk://#diff-c04973e3a3095798cf15be72420af0687bc8cc748028e26fd47b27d4c3c37d5dR73) `web/libs/ui/src/lib/select/select.tsx` - [[2]](diffhunk://#diff-567459e612a12d17fb4237f0381f79f4148ad291939afbf005a1c02e398fe942R77)

* **Integrated `displayValueOverride` in Rendering Logic**: Updated the rendering logic of the `Select` component to use the `displayValueOverride` function if provided. If not, the default rendering logic is applied. (`web/libs/ui/src/lib/select/select.tsx` - [[1]](diffhunk://#diff-567459e612a12d17fb4237f0381f79f4148ad291939afbf005a1c02e398fe942R223-R224) [[2]](diffhunk://#diff-567459e612a12d17fb4237f0381f79f4148ad291939afbf005a1c02e398fe942R246-R247)

### Storybook Updates:

* **Added Examples for `displayValueOverride`**: New examples were added to the `Select` component's Storybook to demonstrate the usage of `displayValueOverride`. These examples showcase custom display logic, such as appending text to selected options or overriding the display value entirely. (`web/libs/storybook/select/select.stories.tsx` - [web/libs/storybook/select/select.stories.tsxR146-R173](diffhunk://#diff-d276d7c9492791bd2cd5819bc7d8227747ea5b3df3a30ecb0b749f97b86b27d8R146-R173))

yyassi-heartex committed:
feat: OPTIC-2161: Give Select a way to override the display value completely

yyassi-heartex committed:
lint clean up

yyassi-heartex commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14866569167)
> Successfully merged:  create mode 100644 web/libs/core/src/lib/utils/query-client.ts

yyassi-heartex committed:
Merge branch 'develop' into 'fb-optic-2161/display-override'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14866569167

nass600 commented on web/libs/ui/src/lib/select/select.tsx:
```diff
@@ -219,26 +220,32 @@ export const Select = forwardRef(
-                      </span>
-                    );
-                  })}
+                  {selectedOptions?.length ? (
```
I would move this fragment to a separate function to make it cleaner

nass600 commented on web/libs/storybook/select/select.stories.tsx:
```diff
@@ -143,6 +143,34 @@ const meta: Meta<typeof Select> = {
+              { label: "No", value: false },
+            ]}
+            label="Boolean values with displayValueOverride"
+            displayValueOverride={(selectedOptions, placeholder) => {
```
I would drop the `Override` bit of the prop to make it cleaner and less redundant

yyassi-heartex replied:
the overRide is to indicate that its a function - instead of just a `displayValue` which indicate its a `string`, or some sort of html element

yyassi-heartex replied:
could do displayValueHandler instead?

nass600 replied:
`renderSelected` is great, thanks

AndrejOros commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14881184509)

AndrejOros committed:
Merge branch 'develop' into 'fb-optic-2161/display-override'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14881184509

AndrejOros committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14881184509

yyassi-heartex committed:
slight refactor

yyassi-heartex committed:
displayValueOverride to renderSelected
